summingbird-storm-test/src/test/scala/com/twitter/summingbird/storm/TopologyTests.scala

```diff
@@ -101,6 +101,109 @@ class TopologyTests extends WordSpec {
+      .map(x => (x,1)).name(optNodeName)
+      .sumByKey(TestStore.createStore[Int, Int]()._2).name(smNodeName)
+
+    val opts = Map(optNodeName -> Options().set(FMMergeableWithSource(true)).set(FlatMapParallelism(5)),
+      sourceNodeName -> Options().set(SourceParallelism(10)),
+      smNodeName -> Options().set(SummerParallelism(7)))
+    val storm = Storm.local(opts)
+    val stormTopo = storm.plan(p).topology
+    val bolts = stormTopo.get_bolts
+    val spouts = stormTopo.get_spouts
+
+    assert(stormTopo.get_bolts_size()==1 && stormTopo.get_spouts_size()==1)
+    assert(spouts.head._2.get_common().get_parallelism_hint()==10)
+    assert(bolts("Tail").get_common().get_parallelism_hint()==7)
+  }
+
+  /*
+ *  Test Case to check if it errors out with a RuntimeException when the FMMergeableWithSource is opted-in and sourceParallelism is smaller than FMParallelism
+ *  Asserts : RuntimeException has occurred when sourceParallelism is smaller than FM Parallelism
+ */
+  "While FMMergeableWithSource Option is set to true, error out when SourceParallelism is less that FMParallelism" in {
+    val fmNodeName = "flatMapper"
+    val smNodeName = "summer"
+    val sourceNodeName = "source"
+    val p = Storm.source(TraversableSpout(sample[List[Int]])).name(sourceNodeName)
+      .flatMap(testFn).name(fmNodeName)
+      .sumByKey(TestStore.createStore[Int, Int]()._2).name(smNodeName)
+
+    val opts = Map(fmNodeName -> Options().set(FMMergeableWithSource(true)).set(FlatMapParallelism(15)),
+      sourceNodeName -> Options().set(SourceParallelism(10)),
+      smNodeName -> Options().set(SummerParallelism(7)))
+    val storm = Storm.local(opts)
+    try{
+      val stormTopo = storm.plan(p).topology
+      assert(false)
+    }
+    catch{
+      case _ : RuntimeException => assert(true)
+    }
+  }
+
+  /*
+  * Test : FlatMap has a fanout and the producer has a opt-in FMMergeableWithSource.
+  * Behavior : When a flatMap has a fanOut then the FlatMappedProducer should not go into Source.
+  * Asserts : Number of bolts : flatMap, Map, Summer1, Summer2
+  *           Number of spouts : source.
+  */
+  "FMMergeableWithSource with a fanOut case after flatMap" in {
+    val sumName = "summer"
+    val p1 = Storm.source(TraversableSpout(sample[List[Int]])).flatMap( testFn )
+    val p2 = p1.sumByKey(TestStore.createStore[Int, Int]()._2).name("sum1")
+    val p3 = p1.map{ x => x }.sumByKey(TestStore.createStore[Int, Int]()._2).name("sum2")
+    val p = p2.also(p3)
+    val opts = Map("sum1" -> Options().set(FMMergeableWithSource(true)).set(FlatMapParallelism(15)),
+      "sum2" -> Options().set(SourceParallelism(50)).set(FMMergeableWithSource(true)))
+    val storm = Storm.local(opts)
+    val stormTopo = storm.plan(p).topology
+    val bolts = stormTopo.get_bolts()
+    val spouts = stormTopo.get_spouts()
+    assert(bolts.size() == 4) // FlatMap should go into SourceNode when it has a fanOut.FlatMap, Map, Summer, Summer
```

Comment:
this is good, but are we sure it plans it correctly? For instance, can we check that the shape is what we expect?